### ISSUE TITLE
Make software.name free text

### ISSUE THREAD

**Issue by SuperTux88** [jhass/nodeinfo#15]
I planned to validate and use the data from NodeInfo in the diaspora ConnectionTester to detect if a server is able to receive diaspora-messages (If it is really a Pod or a compatible other network, or if it is only a website). But after [this comment](https://github.com/diaspora/diaspora/pull/7120#issuecomment-249439678) I realized that this is not possible (with the current NodeInfo version), because `software.name` is locked to a fixed set of names, and can only be extended in a new version of NodeInfo. So if a new software implements the diaspora-protocol, it wont be able to communicate with diaspora until every pod updated to a newer NodeInfo version, even if the protocol itself is compatible.

In my opinion the `software.name` is more like a "User-Agent"-field and it would be weird if every webserver needs to be updated every time a new browser is created, and otherwise the new browsers wouldn't be able to communicate with the servers even if they speak http.

@jaywink tried to do this in #12, but he also changed the `protocols` and `services`. But they should stay enums, so they can be validated.

So the target of loosening `software.name` is that a new software can tell existing software that it talks the same protocols (protocols already known by existing software). If the new software also (additionally) talks the `fancyNewProtocol`, it should only include allowed protocols in the current version of NodeInfo. Existing software wouldn't understand the `fancyNewProtocol` anyway, so there is no disadvantage when omitting it in existing versions of NodeInfo. `fancyNewProtocol` can be added to the next version of NodeInfo and every other software is able to tell that it also talks `fancyNewProtocol` now by updating NodeInfo (and implementing the new protocol).

**Comment by jaywink** [comment]
This would be a nice improvement 👍 

I still fail to see why `protocols` and `services` should be locked as enums in the spec. It's basically over-engineering and trying to control something that should provide a way to *identify* freely what the node is and can do. Whether the reader can understand that information is irrelevant and hardly worth trying to lock down in a specification.

Let's say NodeInfo 1.2 is out. Along comes `fancyProtocol` which a bunch of sites start supporting. Not only has NodeInfo itself have to bump versions but everybody who uses NodeInfo has to then change their NodeInfo implementation (keeping old versions, of course!). So to implement `fancyProtocol`, site A and site B need to first get the string `fancyProtocol` into NodeInfo and then both implement the new NodeInfo in their software stacks.

Of course they can just implement the protocol and advertise by some easier way to each other that they speak `fancyProtocol`. Which is what NodeInfo *could* be good for.

Just my 2 cents :)

**Comment by SuperTux88** [comment]
> should provide a way to identify freely what the node is and can do.

And what is the benefit that a software can tell that it supports `fancyProtocol` if everybody else doesn't know that protocol anyway?

> everybody who uses NodeInfo has to then change their NodeInfo implementation

No. Only a software which wants to support `fancyProtocol` needs updating NodeInfo. For everybody else the old Version is still doing the job.

> So to implement `fancyProtocol`, site A and site B need to first get the string `fancyProtocol` into NodeInfo

Adding a string to the protocols list can be done in a few minutes to a few days (depending of the time that @jhass has), but that is no problem. (I think you need more time to implement the actual protocol ;) )

But anyway, this issue is about changing `software.name`, so a new software can announce **existing** protocols. If you want to change `protocols`, please open a new issue.

**Comment by jaywink** [comment]
I've already submitted a PR which was declined - no need to ask me to do that again :) I only replied here because you yourself wrote about `services` and `protocols` in the description, quoting me and explaining why my opinion was wrong.

Sorry to provide noise. Anyway, just one more comment, I promise I wont add any more discussion here or reply unless asked to..

> And what is the benefit that a software can tell that it supports fancyProtocol if everybody else doesn't know that protocol anyway?

And what is the benefit of including those strings as enum's in the spec? NodeInfo is only doing it because it can - there is no benefit. It's just over engineering a simple problem. Don't solve problems that are not there, solve problems that are real. I'm sure there is a quote for that :)

Anyhow, signing out ✌️ 👋
